packages/indy-vdr/tests/indy-vdr-did-registrar.e2e.test.ts

```diff
@@ -0,0 +1,348 @@
+import {
+  Key,
+  InjectionSymbols,
+  CacheModuleConfig,
+  InMemoryLruCache,
+  JsonTransformer,
+  KeyType,
+  SigningProviderRegistry,
+  TypedArrayEncoder,
+  DidCommV1Service,
+  DidCommV2Service,
+  DidDocumentService,
+  IndyWallet,
+} from '@aries-framework/core'
+import { convertPublicKeyToX25519, generateKeyPairFromSeed } from '@stablelib/ed25519'
+import { Subject } from 'rxjs'
+
+import { IndyStorageService } from '../../core/src/storage/IndyStorageService'
+import { agentDependencies, getAgentConfig, getAgentContext } from '../../core/tests/helpers'
+import testLogger from '../../core/tests/logger'
+import { IndyVdrIndyDidRegistrar } from '../src/dids/IndyVdrIndyDidRegistrar'
+import { IndyVdrIndyDidResolver } from '../src/dids/IndyVdrIndyDidResolver'
+import { indyDidFromNamespaceAndInitialKey } from '../src/dids/didIndyUtil'
+import { IndyVdrPoolService } from '../src/pool/IndyVdrPoolService'
+import { DID_INDY_REGEX } from '../src/utils/did'
+
+import { indyVdrModuleConfig } from './helpers'
+
+const logger = testLogger
+const wallet = new IndyWallet(agentDependencies, logger, new SigningProviderRegistry([]))
+
+const agentConfig = getAgentConfig('IndyVdrIndyDidRegistrar E2E', { logger })
+
+const cache = new InMemoryLruCache({ limit: 200 })
+const indyVdrIndyDidResolver = new IndyVdrIndyDidResolver()
+const indyVdrIndyDidRegistrar = new IndyVdrIndyDidRegistrar()
+
+let signerKey: Key
+
+const agentContext = getAgentContext({
+  wallet,
+  agentConfig,
+  registerInstances: [
+    [InjectionSymbols.Stop$, new Subject<boolean>()],
+    [InjectionSymbols.AgentDependencies, agentDependencies],
+    [InjectionSymbols.StorageService, new IndyStorageService(agentDependencies)],
+    [IndyVdrPoolService, new IndyVdrPoolService(logger, indyVdrModuleConfig)],
+    [CacheModuleConfig, new CacheModuleConfig({ cache })],
+  ],
+})
+
+const indyVdrPoolService = agentContext.dependencyManager.resolve(IndyVdrPoolService)
+
+describe('Indy VDR registrar E2E', () => {
+  beforeAll(async () => {
+    await indyVdrPoolService.connectToPools()
+
+    if (agentConfig.walletConfig) {
+      await wallet.createAndOpen(agentConfig.walletConfig)
+    }
+
+    signerKey = await wallet.createKey({
+      seed: '000000000000000000000000Trustee9',
+      keyType: KeyType.Ed25519,
+    })
+  })
+
+  afterAll(async () => {
+    for (const pool of indyVdrPoolService.pools) {
+      pool.close()
+    }
+
+    await wallet.delete()
+  })
+
+  test('can register a did:indy without services', async () => {
+    const didRegistrationResult = await indyVdrIndyDidRegistrar.create(agentContext, {
+      method: 'indy',
+      options: {
+        submitterDid: 'did:indy:pool:localtest:TL1EaPFCZ8Si5aUrqScBDt',
+        submitterVerkey: signerKey.publicKeyBase58,
+      },
+    })
+
+    expect(JsonTransformer.toJSON(didRegistrationResult)).toMatchObject({
+      didDocumentMetadata: {
+        qualifiedIndyDid: expect.stringMatching(DID_INDY_REGEX),
```

Comment:
We don't really need this anymore I guess?

Reply:
Yes, in this case makes no sense as full did is available within the doc